u2f-ref-code/java/src/com/google/u2f/server/messages/RegisteredKey.java

```diff
@@ -117,9 +120,9 @@ public JsonObject getJson() {
     result.addProperty("version", version);
     result.addProperty("keyHandle", keyHandle);
     result.addProperty("sessionId", sessionId);
-    JsonArray transportsJson = getTransportsAsJson();
-    if (transportsJson != null) {
-      result.add("transports", transportsJson);
+    String transportsString = getTransportsAsString();
+    if (transportsString != null) {
```

Comment:
You need to test for empty string here I think.

```String.join(",", <empty list>)``` will return an empty string -- just verified.